tests/codex/testproving.nim

```diff
@@ -49,7 +49,7 @@ suite "Proving":
     proving.onProofRequired = onProofRequired
     proofs.setProofRequired(id, true)
     await proofs.advanceToNextPeriod()
-    check called
+    check eventually called
```

Comment:
Out of curiosity, what does adding the `eventually` solve? Why do we sometimes need to wait 5 seconds?

Reply:
The call above to `advanceToNextPeriod()` used to include a 2 second delay. I removed that delay, so that we can now wait the exact time that it takes for the callback to be called.